### PR TITLE
chore(NODE-6578): add misc tooling to ZSTD repo

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

This PR contains a smattering of tooling we'll need for the SSDLC release flow (coming in the next PR):

- A release please config and manifest has been added.
- A new `build` action has been added, that builds the bindings and uploads the to GHA artifact storage (for use in the release action later, coming in a follow-up PR)
- CodeQL has been enabled and configured for both C++ and JS.
- A new SBOM file has been generated, which includes the correct version of zstd as our only bundled dependency.
- The zstd C library version now is specified in the package.json and the install scripts reads the package file to install the correct version of zstd.

### Double check the following

- [ ] Ran `npm run format:js && npm run format:rs` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    runs-on: "ubuntu-latest"
+    timeout-minutes: 360
```
Comment (aditi-khare-mongoDB) [inline .github/workflows/codeql.yml:13]: CodeQL only takes about 2 minutes, is there a reason we put a timeout here and chose 360?

File: .github/docker/Dockerfile.musl [hunk 2]
```diff
@@ -2,13 +2,19 @@
 ARG PLATFORM=arm64
 ARG NODE_VERSION=16.20.1
 
-FROM ${PLATFORM}/node:${NODE_VERSION}-alpine AS node
+FROM ${PLATFORM}/node:${NODE_VERSION}-alpine AS build
```
Comment (aditi-khare-mongoDB) [inline .github/docker/Dockerfile.musl:5]: What's the motivation behind this change?

Comment (baileympearson) [reply .github/docker/Dockerfile.musl:5]: The last line in the dockerfile:
```
COPY --from=build /zstd/prebuilds/ /
```

Requires that tag used in --from is the same as the tag here.  I chose to change this to build to match Dockerfile.glibc instead of adjusting the tag below to `--from=node`.

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (aditi-khare-mongoDB) [reply .github/workflows/codeql.yml:13]: No need to change it if it matches the existing code for mongodb-client-encryption!

**Review by aditi-khare-mongoDB — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -49,5 +50,6 @@
     "napi_versions": [
       4
     ]
-  }
-}
+  },
+  "zstd_version": "1.5.6"
```
Comment (nbbeeken) [inline package.json:54]: It's highly improbable that this would ever clash with existing tooling but to be beyond certain that this is our field for us to manage can we "namespace" it with a prefix like we have in [mongodb-client-encryption](https://github.com/mongodb-js/mongodb-client-encryption/blob/main/package.json#L38) and [bson](https://github.com/mongodb/js-bson/blob/main/package.json#L91)

**Review by nbbeeken — CHANGES_REQUESTED** [review]
> Warning:   0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override

Why doesn't this happen on our other repos?

**Comment by baileympearson** [comment]
> Warning: 0:0 warning File ignored by default. Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override
> 
> Why doesn't this happen on our other repos? I'm seeing github post a warning on the .release-please-manifest.json file

We don't lint json files in our other repos.  I set this up when I set up eslint and mocha - you pointed out they were malformed and we decided to lint them.

**Review by W-A-James — APPROVED** [review]
